audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractRestTest.java

```diff
@@ -0,0 +1,153 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.util.LinkedMultiValueMap;
+
+import uk.gov.gchq.palisade.service.audit.AuditApplication;
+import uk.gov.gchq.palisade.service.audit.config.AuditServiceConfigProperties;
+import uk.gov.gchq.palisade.service.audit.service.AuditService;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.springframework.http.HttpStatus.ACCEPTED;
+import static uk.gov.gchq.palisade.contract.audit.ContractTestData.BAD_SUCCESS_REQUEST_OBJ;
+import static uk.gov.gchq.palisade.contract.audit.ContractTestData.ERROR_REQUEST_OBJ;
+import static uk.gov.gchq.palisade.contract.audit.ContractTestData.GOOD_SUCCESS_REQUEST_OBJ;
+import static uk.gov.gchq.palisade.contract.audit.ContractTestData.REQUEST_TOKEN;
+import static uk.gov.gchq.palisade.service.audit.model.Token.HEADER;
+
+/**
+ * An external requirement of the service is to connect to a pair of upstream kafka topics.
+ * <ol>
+ *     <li>The "error" topic can be written to by any service that encounters an error when processing a request</li>
+ *     <li>The "success" topic should only be written to by the filtered-resource-service or the data-service</li>
+ * </ol>
+ * This service does not write to a downstream topic
+ */
+@SpringBootTest(
+        classes = AuditApplication.class,
+        webEnvironment = WebEnvironment.RANDOM_PORT,
+        properties = {"akka.discovery.config.services.kafka.from-config=false"}
+)
+@Import(KafkaInitializer.Config.class)
+@ContextConfiguration(initializers = {KafkaInitializer.class})
+@ActiveProfiles({"akka-test"})
+class KafkaContractRestTest {
+
+    static final Logger LOGGER = LoggerFactory.getLogger(KafkaContractRestTest.class);
+
+    @Autowired
+    AuditServiceConfigProperties auditServiceConfigProperties;
+
+    @Autowired
+    TestRestTemplate restTemplate;
+
+    @SpyBean
+    private AuditService auditService;
+
+    @AfterEach
+    void tearDown() {
+        Arrays.stream(new File(auditServiceConfigProperties.getErrorDirectory()).listFiles())
+            .filter(file -> (file.getName().startsWith("Success") || file.getName().startsWith("Error")))
+            .peek(file -> LOGGER.info("Deleting file {}", file.getName()))
+            .forEach(File::deleteOnExit);
+    }
+
+    @Test
+    @DirtiesContext
+    void testRestEndpointForErrorMessage() throws InterruptedException {
```

Comment:
I think that the general best practice is declare test methods to throw Exception and not specific sub-types.

Reply:
Updated